src/Features/ShippingLabelBannerDisplayRules.php

```diff
@@ -0,0 +1,237 @@
+	 */
+	private $min_jetpack_version = '4.4';
+
+	/**
+	 * Minimum supported WooCommerce Services version.
+	 *
+	 * @var string
+	 */
+	private $min_wcs_version = '1.22.5';
+
+	/**
+	 * Supported countries by USPS, see: https://webpmt.usps.gov/pmt010.cfm
+	 *
+	 * @var array
+	 */
+	private $supported_countries = array( 'US', 'AS', 'PR', 'VI', 'GU', 'MP', 'UM', 'FM', 'MH' );
+
+	/**
+	 * Array of supported currency codes.
+	 *
+	 * @var array
+	 */
+	private $supported_currencies = array( 'USD' );
+
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $jetpack_version Installed Jetpack version to check.
+	 * @param bool   $jetpack_connected Is Jetpack connected?.
+	 * @param string $wcs_version Installed WooCommerce Services version to check.
+	 * @param bool   $wcs_tos_accepted WooCommerce Services Terms of Service accepted?.
+	 */
+	public function __construct( $jetpack_version, $jetpack_connected, $wcs_version, $wcs_tos_accepted ) {
+		$this->jetpack_version   = $jetpack_version;
+		$this->jetpack_connected = $jetpack_connected;
+		$this->wcs_version       = $wcs_version;
+		$this->wcs_tos_accepted  = $wcs_tos_accepted;
+	}
+
+	/**
+	 * Determines whether or not the banner should be displayed.
+	 */
+	public function should_display_banner() {
+		return $this->banner_not_dismissed() &&
+			$this->jetpack_installed_and_active( $this->jetpack_version ) &&
+			$this->jetpack_up_to_date( $this->jetpack_version ) &&
+			$this->jetpack_connected &&
+			$this->ups_not_installed() &&
+			$this->fedex_not_installed() &&
+			$this->shippingeasy_not_installed() &&
+			$this->shipstation_not_installed() &&
+			$this->order_has_shippable_products() &&
+			$this->store_in_us_and_usd() &&
+			( $this->wcs_not_installed() || (
+				$this->wcs_up_to_date() && ! $this->wcs_tos_accepted
+			) );
+	}
+
+	/**
+	 * Checks if the banner was not dismissed by the user.
+	 *
+	 * @return bool
+	 */
+	private function banner_not_dismissed() {
+		$dismissed_timestamp = get_option( 'woocommerce_shipping_dismissed_timestamp' );
+
+		$dismissed_for_good = -1 === $dismissed_timestamp;
+		$dismissed_24h      = time() < $dismissed_timestamp;
+
+		return ! $dismissed_for_good && ! $dismissed_24h;
+	}
+
+	/**
+	 * Checks if jetpack is installed and active.
+	 *
+	 * @param string $jetpack_version Jetpack version to check.
+	 * @return bool
+	 */
+	private function jetpack_installed_and_active( $jetpack_version ) {
+		return ! ! $jetpack_version;
+	}
+
+	/**
+	 * Checks if Jetpack version is supported.
+	 *
+	 * @param string $jetpack_version Installed Jetpack version to check.
+	 * @return bool
+	 */
+	private function jetpack_up_to_date( $jetpack_version ) {
+		return version_compare( $jetpack_version, $this->min_jetpack_version, '>=' );
+	}
+
+	/**
+	 * Checks if Fedex shipping method is not installed.
+	 *
+	 * @return bool
+	 */
+	private function fedex_not_installed() {
+		return ! is_plugin_active( 'woocommerce-shipping-fedex/woocommerce-shipping-fedex.php' );
```

Comment:
Using the path/slugs to determine if a plugin is active or not doesn't cover the use case of a plugin being installed in a differently named directory. Totally an edge case, but if possible it is nice to check for existence of constant defined in the plugin, or a unique method name.

Reply:
Thanks for the tip! I replaced all `is_plugin_active` statements with `class_exists`.